app/src/main/java/com/example/stromtracker/ui/geraete/GeraeteFragment.kt

```diff
@@ -1,125 +1,186 @@
-
-
-
-    /*
-        geraeteViewModel.setGereateList(Geraete(1, "test", "test2"))
-        Log.d("TAG2", geraeteViewModel.getGereateList().toString())
-
-
-     */
-
-
-
-
-
-
-
+        root = inflater.inflate(R.layout.fragment_geraete, container, false)
+        buttonAddVerbraucher = root.findViewById(R.id.button_geraete_add_verbraucher)
+        buttonAddVerbraucher.setOnClickListener(this)
+        buttonAddProduzent = root.findViewById(R.id.button_geraete_add_produzent)
+        buttonAddProduzent.setOnClickListener(this)
+
+        buttonSortVerbrauch = root.findViewById(R.id.geraete_button_sort_verbrauch)
+        buttonSortVerbrauch.setOnClickListener(this)
+        buttonSortRaum = root.findViewById(R.id.geraete_button_sort_raum)
+        buttonSortRaum.setOnClickListener(this)
+        buttonSortName = root.findViewById(R.id.geraete_button_sort_name)
+        buttonSortName.setOnClickListener(this)
+
+        geraeteList = ArrayList()
+        kategorieList = ArrayList()
+        raumList = ArrayList()
+
+
+        viewManager = LinearLayoutManager(this.context)
+        viewAdapter = GeraeteListAdapter(geraeteList, kategorieList, raumList)
+        recyclerView = root.findViewById<RecyclerView>(R.id.geraete_recycler_view).apply {
+            setHasFixedSize(true)
+            layoutManager = viewManager
+            adapter = viewAdapter
+        }
         return root
     }
 
+
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
 
-        //geraeteViewModel= ViewModelProvider(this).get(GeraeteViewModel::class.java)
+
 
         geraeteViewModel = ViewModelProviders.of(this).get(GeraeteViewModel::class.java)
 
-        geraeteViewModel.getAllGeraete().observe(
+        geraeteViewModel.getAllVerbraucher().observe(
             viewLifecycleOwner,
             Observer { geraete ->
                 if (geraete != null) {
+                    geraeteList.clear()
+                    geraeteList.addAll(geraete)
+                    viewAdapter.notifyDataSetChanged();
 
-                    setGeraeteList(geraete)
-                    Log.d("TAGGeraete", geraete.toString())
+                }
+            })
 
-                    if (geraete.isEmpty()) {
+        geraeteViewModel.getAllRaeume().observe(
+            viewLifecycleOwner,
+            Observer { raeume ->
+                if (raeume != null) {
+                    raumList.clear()
+                    raumList.addAll(raeume)
+                    if(raeume.isEmpty()) {
+                        //nur zum testen
                         geraeteViewModel.insertHaushalt(Haushalt("name", 0.0, 1, 0.0, null, false))
+                        geraeteViewModel.insertRaum(Raum("test", 1))
+                        geraeteViewModel.insertRaum(Raum("zet", 1))
                         geraeteViewModel.insertKategorie(Kategorie("test", null))
-                        geraeteViewModel.insertRaum(Raum("test"))
```

Comment:
Das sollten wir auf jeden Fall noch in die MainActivity einfügen, da wir sonst eine Inkosistente Datenbank bekommen, wenn der User beim ersten Aufruf z.B. erst die Geräte Seite aufruft. Bzwe. ist hier das Problem, dass seöbst wenn ein Haushalt existiert, hier nochmal einer erstellt wird, und dann aber dem alten Haushalt die Geräte hinzugefügt werden.

Aber das ist etwas, dass wir noch in allen Fragments machen müssen, die auf die Datenbank zugreifen.

Reply:
Ist wie gesagt nur zum testen für mich, das ist nichts was bleiben soll.

Reply:
Ja passt.